uaclient/log.py

```diff
@@ -58,3 +60,35 @@ def format(self, record: logging.LogRecord) -> str:
 
         local_log_record["extra"] = extra_message_dict
         return json.dumps(list(local_log_record.values()))
+
+
+def get_user_log_file() -> str:
+    """Gets the correct user log_file storage location
+
+    returns XDG_CACHE_HOME env variable, if it is set,
+    If not set, we return the default path
+    """
+    user_cache = os.environ.get("XDG_CACHE_HOME")
+    log_file = defaults.USER_LOG_FILE_PATH
+    if user_cache:
+        return user_cache + "/" + log_file
+    return pathlib.Path.home().as_posix() + "/.cache/" + log_file
+
+
+def get_all_user_log_files() -> List[str]:
+    """Gets all the log files for the users in the system
+
+    Returns a list of all user log files in their home directories.
+    """
+    user_directories = os.listdir("/home")
+    dirlist = []
```

Comment:
This isn't a list of dirs, so let's not call it a dirlist
```suggestion
    log_files = []
```